pkg/cue/model/value/value.go

```diff
@@ -82,6 +83,30 @@ func (val *Value) UnmarshalTo(x interface{}) error {
 	return json.Unmarshal(data, x)
 }
 
+// SubstituteInStruct substitute expr in struct lit value
+// nolint:staticcheck
+func (val *Value) SubstituteInStruct(new ast.Expr, key string) error {
```

Comment:
The parameter `new` conflicts with the keyword `new`. maybe a different name would be better?